### PR TITLE
Always train with GSE37418

### PR DESCRIPTION
**Pull request by envest**
Purpose: we need to force array study GSE37418 to always be in the training data set since our comparison tool (medulloPackage) uses it to train. We want to avoid data leakage! 💧 

This PR:
- adds a generic option for array or RNA-seq studies to force into training (utils/modeling.R)
- when randomly assigning studies to train/test, `get_train_test_samples()` keeps the same parameter for proportion of studies going to train/test while also accounting for the forced training sets
- includes "GSE37418" in training data in `analysis_notebooks/baseline_models.Rmd`

Questions I have:
- What edge cases need to be accounted for?
- What level of sanity check beyond what's already there is appropriate?

Thanks!

### CODE REVIEW

File: utils/modeling.R [hunk 1]
```diff
@@ -270,9 +278,52 @@ get_train_test_samples <- function(genex_df,
...
+  # actually in metadata
+  always_train_array_error_message <- NULL
+  if (!all(always_train_array %in% array_studies)) {
+
+    missing_array <- always_train_array[which(!(always_train_array %in% array_studies))]
+
+    always_train_array_error_message <- stringr::str_c(
+      "Array studies ",
+      stringr::str_c(missing_array, collapse = ", "),
+      " are required to be in training data but are not found in metadata.")
```
Comment (jaclyn-taroni) [inline utils/modeling.R:291]: Do you want to throw an `\n` at the end of this and the RNA-seq message?

**Review by jaclyn-taroni — APPROVED** [review]
This looks good to me. To answer your questions, I can't think of any cases you haven't addressed with what you've added here.

**Comment by envest** [comment]
Thank you @jaclyn-taroni !
